base/linalg/lq.jl

```diff
@@ -88,22 +88,25 @@ end
 
 size(A::LQPackedQ) = size(A.factors)
 
-full(A::LQ) = A[:L]*A[:Q]
-#=
-We construct the full eye here, even though it seems ineffecient, because
-every element in the output matrix is a function of all the elements of
-the input matrix. The eye is modified by the elementary reflectors held
-in A, so this is not just an indexing operation. Note that in general
-explicitly constructing Q, rather than using the ldiv or mult methods,
-may be a wasteful allocation.
-=#
-function full{T}(A::LQPackedQ{T}; thin::Bool=true)
+convert(::Type{Matrix}, A::LQ) = convert(Array, A[:L]*A[:Q])
+convert(::Type{Array}, A::LQ) = convert(Matrix, A)
+full(A::LQ) = convert(Array, A)
+
+function convert{T}(::Type{Matrix}, A::LQPackedQ{T}; thin::Bool = true)
```

Comment:
Sorry for the silence. I've been thinking quite a bit about what the right thing to do here is. 

Transposed multiplication with a `Q` factor, say `Q'A` defaults to use the square `Q`. This is "the natural" thing to return and is quite useful for least squares. (For non-transposed multiplication, the size of `A` determines how much of `Q` is used.) I think this suggests that `full(Q)` should be square. Otherwise `Q'full(Q)` and `full(Q)'Q` would be rectangular, `full(Q)*Q'` not the identity (could be justified) but `Q*full(Q)'` would be an error. If `full(Q)` is square then `Q'full(Q)=full(Q)'Q=Q*full(Q)'=full(Q)*Q'=I`.

Returning a square `Q` is, of course, a lot of waste but it should be a rare operation to use. If you are interested in the dense version you'd probably start out with using `qr` over `qrfact` anyway and I'd be fine with `qr` still return the thin version of `Q` by default. Consequently, I think that the `full(Q)` methods can be removed.